100_essentials/greetings/README.md

```diff
@@ -0,0 +1,59 @@
+# Greetings
+
+Everything you need to complete this challenge can be found [here](https://docs.python.org/3.8/tutorial/controlflow.html#defining-functions).
+
+---
+
+Define a function named `greet` that takes in a name and a message as inputs and returns a greeting in the following format:
+
+```python
+>>> greet("Mark", "Oh, hi")
+'Oh, hi, Mark!'
+```
+
+Additionally, the message should default to "Hello" if none is provided.
+
+```python
+>>> greet("Dave")
+'Hello, Dave!'
+```
+
+Define a function named `greet_with_style` that takes a name as a required positional parameter, an arbitrary amount of extra positional parameters, an optional key word parameter named `msg` that defaults to 'Hello', another optional key word parameter named `sep` that defaults to ', ', and a required keyword parameter named `punctuation`. `greet_with_style` should return a greeting (name + sep + message + puncuation) with any functions passed as additional positional arguments applied to it.
+
+```python
+>>> def starify(s):
+...    return "*".join(list(s))
+...
+>>> def capitalize(s):
+...    return s.upper()
+...
+>>> greet_with_style("Citizen", starify, capitalize, msg="Greetings", punctuation="!", sep='*')
+'G*R*E*E*T*I*N*G*S***C*I*T*I*Z*E*N*!'
+```
```

Comment:
This is _kind of_ an unspoken rule of the challenges, but in general you want them to define _one_ function/class/whatever (except in advanced cases) and then refactor that function over and over to get new/more flexible results.

Additionally, you essentially never want to show implementation, only behavior.

A way you could edit this here would be to have them add a keyword argument that specifies they'd like to add a separator character, an `lower` or `upper` kwarg that can be a boolean, etc.

Ultimately, I think we should likely save teaching them that functions are first class / that you can pass functions to another function until we get to the 200 or 300 level, and right now should be focused on showing them how to define flexible, composable functions.

Down to discuss this though, it's not a _hard_ requirement, just something I think we should standardize on.